ieeeuoft/hackathon-template
# IEEE 253 fixing order modal cancel bug

## Overview

Description: This PR (for Jira ticket IEEE-253) contains fixes to the order modal cancel bug.

Bug description: Originally, the orders would be canceled chronologically. We don't want that. We want the correct orders to be canceled (corresponding to which cancel button for which order the user presses)

Speculation as to why the bug occurred: The reason why orders were being canceled in chronological order is that in the submitModal() method, the orderID being retrieved is always the most recent orderID (which as a result, would cancel the most recent order, not the order in which the participant originally wanted). I

Summary of changes: Created a new variable called orderId that stores the desired canceled order's id based on the "Cancel order" button press. The state of this variable is updated each time the button is being pressed and should work even when multiple orders are pending. Note: Implementations using event.stopPropagation() did not work, hence why I resorted to this method of change.

## Unit Tests Created

- Did not create nor modify any unit tests


## Steps to QA

- Would appreciate suggestions to improve or rewrite current solution based on best practices. Would also appreciate testing to see whether bug is fully resolved with this solution and/or other ideas as to why the bug originated in the first place.
